src/main/java/seedu/address/logic/commands/DeleteAttendanceCommand.java

```diff
@@ -0,0 +1,167 @@
+            Index moduleClassIndex, Index lessonIndex, Index studentIndex, Week week) {
+        requireAllNonNull(moduleClassIndex, lessonIndex, studentIndex, week);
+
+        this.moduleClassIndex = moduleClassIndex;
+        this.lessonIndex = lessonIndex;
+        this.studentIndex = studentIndex;
+        this.week = week;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Student> lastShownStudentList = model.getFilteredStudentList();
+        List<ModuleClass> lastShownModuleClassList = model.getFilteredModuleClassList();
+
+        if (studentIndex.getZeroBased() >= lastShownStudentList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        if (moduleClassIndex.getZeroBased() >= lastShownModuleClassList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_CLASS_DISPLAYED_INDEX);
+        }
+
+        Student targetStudent = lastShownStudentList.get(studentIndex.getZeroBased());
+        ModuleClass targetModuleClass = lastShownModuleClassList.get(moduleClassIndex.getZeroBased());
+
+        if (lessonIndex.getZeroBased() >= targetModuleClass.getLessons().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_LESSON_DISPLAYED_INDEX);
+        }
+
+        Lesson targetLesson = targetModuleClass.getLessons().get(lessonIndex.getZeroBased());
+
+        if (!targetLesson.getAttendanceRecordList().isWeekContained(week)) {
+            throw new CommandException(Messages.MESSAGE_INVALID_WEEK);
+        }
+
+        Lesson modifiedLesson = createModifiedLesson(targetLesson, targetStudent, week);
+        ModuleClass modifiedModuleClass = createModifiedModuleClass(targetModuleClass, lessonIndex, modifiedLesson);
+        model.setModuleClass(targetModuleClass, modifiedModuleClass);
+
+        String message = String.format(MESSAGE_DELETE_ATTENDANCE_SUCCESS, week, targetStudent.getName(), targetLesson);
+        model.commit(message);
+        return new CommandResult(message);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof DeleteAttendanceCommand // instance of handles nulls
+                && moduleClassIndex.equals(((DeleteAttendanceCommand) other).moduleClassIndex)
+                && lessonIndex.equals(((DeleteAttendanceCommand) other).lessonIndex))
+                && studentIndex.equals(((DeleteAttendanceCommand) other).studentIndex)
+                && week.equals(((DeleteAttendanceCommand) other).week);
+    }
+
+    private static ModuleClass createModifiedModuleClass(ModuleClass targetModuleClass,
+                                                         Index lessonToEditIndex, Lesson lessonToUpdate) {
+        assert targetModuleClass != null;
+        assert lessonToUpdate != null;
```

Comment:
Should you insert an assertion for `lessonToEditIndex`?
```suggestion
        assert targetModuleClass != null;
        assert lessonToEditIndex != null;
        assert lessonToUpdate != null;
```

Reply:
Inserted assertion